subsys/sd/sd.c

```diff
@@ -56,7 +56,7 @@ static int sd_send_interface_condition(struct sd_card *card)
 		resp = cmd.response[0];
 	}
 	if ((resp & 0xFF) != SD_IF_COND_CHECK) {
-		LOG_INF("Legacy card detected, no CMD8 support");
+		LOG_DBG("Legacy card detected, no CMD8 support");
 		/* Retry probe */
 		return SD_RETRY;
```

Comment:
why are we doing a retry anyways here? do we expect the card to magically stop being a legacy card? maybe this calls for a different change than just changing this log level

Reply:
This command is being sent immediately after the reset command, I could see some situation where a card needs extra time to set itself up and doesn't respond to the first few queries.

Reply:
ok, im not gonna block for a change request, but wondering if @danieldegrasse knows an answer about this

Reply:
I believe that the reason I implemented support like this was to follow the way that SPI cards were handled prior to the SDMMC framework being added: https://github.com/zephyrproject-rtos/zephyr/blob/v2.7-branch/drivers/disk/sdmmc_spi.c#L536. I implemented similar retry logic for initialization in the SDMMC framework. However, Linux does not seem to retry CMD8, so it might be safe to remove this. Not really necessary to do as part of this PR, but I am happy to make/test the change on the SD cards I have access to